ui/apps/dashboard/src/components/Layout/SideBar.tsx

```diff
@@ -0,0 +1,25 @@
+'use client';
+
+import { useState, type ReactNode } from 'react';
```

Comment:
This is the main sidebar code unit. It's a client side component because it has to track collapsible state client side and reflect that change instantly. But it can also have nav components as children that are server rendered if necessary.

Reply:
**question**: I'm mostly curious, how does this work with server rendering? Does the logo for example render on the server as you mentioned that the children are rendered on the server? or the collpases/uncollapsed state then just the state only us used during toggles?

Reply:
Sidebar is a client component because it has a state hook to set and reflect the collapse instantly. Client component are still pre-rendered on the server so the logo does not pop in later or anything. I also pass a mix of server and client components as children to the sidebar based on whether or not they have client-only functionality. See here for more on this client <> composition patter: https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props. 

Check out the new screen recording now that I've merged the env fetch refactor, you'll only the env menu renders in after client side data fetches.